org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/contentassist/GetterSetterCompletionProposal.java

```diff
@@ -0,0 +1,138 @@
+/*******************************************************************************
+ * Copyright (c) 2000, 2016 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Originally copied from org.eclipse.jdt.internal.ui.text.java.GetterSetterCompletionProposal
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+
+package org.eclipse.jdt.ls.core.internal.contentassist;
+
+import java.util.Collection;
+
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.jdt.core.CompletionProposal;
+import org.eclipse.jdt.core.Flags;
+import org.eclipse.jdt.core.IField;
+import org.eclipse.jdt.core.IMethod;
+import org.eclipse.jdt.core.IType;
+import org.eclipse.jdt.core.JavaModelException;
+import org.eclipse.jdt.core.Signature;
+import org.eclipse.jdt.core.dom.rewrite.ImportRewrite;
+import org.eclipse.jdt.core.formatter.CodeFormatter;
+import org.eclipse.jdt.internal.codeassist.InternalCompletionProposal;
+import org.eclipse.jdt.internal.core.manipulation.util.Strings;
+import org.eclipse.jdt.internal.corext.util.CodeFormatterUtil;
+import org.eclipse.jdt.internal.corext.util.JdtFlags;
+import org.eclipse.jdt.ls.core.internal.corext.codemanipulation.GetterSetterUtil;
+import org.eclipse.jface.text.BadLocationException;
+import org.eclipse.jface.text.IDocument;
+import org.eclipse.jface.text.TextUtilities;
+
+/**
+ * Generates GetterSetter completion proposals.
+ */
+public class GetterSetterCompletionProposal extends InternalCompletionProposal {
+
+	private IField fField;
+
+	private boolean fIsGetter;
+
+	public static void evaluateProposals(IType type, String prefix, int offset, int length, int relevance, Collection<CompletionProposal> result) throws JavaModelException {
+		if (prefix.length() == 0) {
+			relevance--;
+		}
+
+		IField[] fields= type.getFields();
+		IMethod[] methods= type.getMethods();
+		for (int i= 0; i < fields.length; i++) {

```

Comment:
generally prefer enhanced for loops when possible.
`for (IField curr : fields)`